### PR TITLE
fix: CUDOS delegations movement fix

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed Changes
When you iterate over `OrderedMap.Keys()` and during iteration call `Delete` the original reference of keys gets corrupted due to in-place append in `Delete`

I've made `Delete` not to modify the original data so now it is safe to do.

_[describe the changes here...]_

## Linked Issues

_[if applicable, add links to issues resolved by this PR]_

## Types of changes

_What type of change does this pull request make (put an `x` in the boxes that apply)?_

- [ ] Bug fix (non-breaking change that fixes an issue).
- [ ] New feature added (non-breaking change that adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected).
- [ ] Documentation update.
- [ ] Something else (e.g., tests, scripts, example, deployment, infrastructure).

## Checklist

_Put an `x` in the boxes that apply:_

- [ ] I have read the [CONTRIBUTING](https://github.com/fetchai/fetchd/blob/main/CONTRIBUTING.md) guide
- [ ] Checks and tests pass locally

### If applicable

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease
- [ ] I have added/updated the documentation

## Further comments

_[if this is a relatively large or complex change, kick off a discussion by explaining why you chose the solution you did, what alternatives you considered, etc...]_

### CODE REVIEW

File: app/upgrade_cudos.go [hunk 1]
```diff
@@ -2032,9 +2032,11 @@ func DoGenesisAccountMovements(genesisData *GenesisData, cudosCfg *CudosMergeCon
 
 		// Handle delegations movement
 		remainingAmountToMove := sdk.NewIntFromBigInt(accountMovement.Amount.BigInt())
+
 		if sourceDelegations, exists := genesisData.Delegations.Get(accountMovement.SourceAddress); exists {
-			for i := range sourceDelegations.Iterate() {
-				validatorAddr, delegatedAmount := i.Key, i.Value
+			// We iterate and delete from source array at the same time
+			for _, validatorAddr := range sourceDelegations.Keys() {
```
Comment (pbukva) [inline app/upgrade_cudos.go:2038]: Are we sure this is the only place (= internals of the whole cycle scope) from which we call delete in the OrderedMap?

Comment (MissingNO57) [reply app/upgrade_cudos.go:2038]: Yes, but Keys() is now safe to use anyway.

Comment (MissingNO57) [reply app/upgrade_cudos.go:2038]: `moveGenesisDelegation` is using `Delete` and `moveGenesisDelegation` this is the only place.

**Review by pbukva — APPROVED** [review]
LGTM

**Pull request merged**